airflow/api/common/experimental/trigger_dag.py

```diff
@@ -68,15 +68,17 @@ def _trigger_dag(
                     execution_date.isoformat(),
                     min_dag_start_date.isoformat()))
 
+    run_type: Optional[DagRunType] = None
     if not run_id:
-        run_id = f"{DagRunType.MANUAL.value}__{execution_date.isoformat()}"
-
-    dag_run_id = dag_run.find(dag_id=dag_id, run_id=run_id)
-    if dag_run_id:
-        raise DagRunAlreadyExists("Run id {} already exists for dag id {}".format(
-            run_id,
-            dag_id
-        ))
+        run_type = DagRunType.MANUAL
+        dag_run = dag_run.find(dag_id=dag_id, run_type=run_type, execution_date=execution_date)
```

Comment:
Is there is already a scheduled dag_run for this exact time, this `find()` will fail to find anything, but the unique constraint (on exeuction_date, dag_id) would still be violated.

Reply:
I will fix it